static/js/actions/ui.js

```diff
@@ -22,9 +23,12 @@ export const setDialogVisibility = (bool: boolean): Action => (
 
 // work history actions
 export const SET_WORK_HISTORY_EDIT = 'SET_WORK_HISTORY_EDIT';
-export const setWorkHistoryEdit = (bool: boolean): Action => (
-  { type: SET_WORK_HISTORY_EDIT, payload: bool }
-);
+export const setWorkHistoryEdit: Function = (bool: boolean): (d: Dispatch) => Promise => {
```

Comment:
the declaration reads (in English) something like:

`setWorkHistoryEdit` is a function. it accepts an argument `bool`, of type `boolean`, and returns a function which takes an argument `d`, of type `Dispatch`, and returns a promise.

it's a little confusing because the annotation for function types looks like `(arg: argType) => returnType`, but if you're annotating an arrow function literal you do `(arg: argType): returnType => { ... }`.